README.md
# README

A vscode theme using a large selection of colors and font styles.
The color palette is based off of the colors in [Chakra UI](https://chakra-ui.com/)

![Preview of syntax highlighting](marketplace/screenshot-1.png?raw=true)


CHANGELOG.md
# Change Log

## [0.0.8]

- Adds custom colorized bracket pair matching
- Darkens terminal and editor breadcrumb backgrounds

## [0.0.7]

- Major changes to increase contrast and reduce usage of reds
- Darkened background color
- Variables now use green variants instead of reds
- Literals now use teal variants instead of greeens
- Types now use orange variants instead of teals

## [0.0.6]

- Altered class member styles

## [0.0.5]

- Improved JSX styles

## [0.0.4]

- Improved SQL highlighting

## [0.0.3]

- Un-bolds object keys and imports

## [0.0.2]

- Reduces required version

## [0.0.1]

- Initial release


scripts/palette.js
const baseColors = {
  transparent: "#00000000",
  black: "#000000",
  white: "#FFFFFF",

  gray: {
    50: "#F7FAFC",
    100: "#EDF2F7",
    200: "#E2E8F0",
    300: "#CBD5E0",
    400: "#A0AEC0",
    500: "#718096",
    600: "#4A5568",
    700: "#2D3748",
    800: "#1A202C",
    900: "#171923",
    1000: "#0F1016",
  },

  red: {
    50: "#FFF5F5",
    100: "#FED7D7",
    200: "#FEB2B2",
    300: "#FC8181",
    400: "#F56565",
    450: "#EA4B4B",
    500: "#E53E3E",
    600: "#C53030",
    700: "#9B2C2C",
    800: "#822727",
    900: "#63171B",
  },

  orange: {
    50: "#FFFAF0",
    100: "#FEEBC8",
    200: "#FBD38D",
    300: "#F6AD55",
    400: "#ED8936",
    500: "#DD6B20",
    600: "#C05621",
    700: "#9C4221",
    800: "#7B341E",
    900: "#652B19",
  },

  yellow: {
    50: "#FFFFF0",
    100: "#FEFCBF",
    200: "#FAF089",
    300: "#F6E05E",
    400: "#ECC94B",
    500: "#D69E2E",
    600: "#B7791F",
    700: "#975A16",
    800: "#744210",
    900: "#5F370E",
  },

  green: {
    50: "#F0FFF4",
    100: "#C6F6D5",
    200: "#9AE6B4",
    300: "#68D391",
    400: "#48BB78",
    500: "#38A169",
    600: "#2F855A",
    700: "#276749",
    800: "#22543D",
    900: "#1C4532",
  },

  teal: {
    50: "#E6FFFA",
    100: "#B2F5EA",
    200: "#81E6D9",
    300: "#4FD1C5",
    400: "#38B2AC",
    500: "#319795",
    600: "#2C7A7B",
    700: "#285E61",
    800: "#234E52",
    900: "#1D4044",
  },

  blue: {
    50: "#ebf8ff",
    100: "#bee3f8",
    200: "#90cdf4",
    300: "#63b3ed",
    400: "#4299e1",
    500: "#3182ce",
    600: "#2b6cb0",
    700: "#2c5282",
    800: "#2a4365",
    900: "#1A365D",
  },

  cyan: {
    50: "#EDFDFD",
    100: "#C4F1F9",
    200: "#9DECF9",
    300: "#76E4F7",
    400: "#0BC5EA",
    500: "#00B5D8",
    600: "#00A3C4",
    700: "#0987A0",
    800: "#086F83",
    900: "#065666",
  },

  purple: {
    50: "#FAF5FF",
    100: "#E9D8FD",
    200: "#D6BCFA",
    300: "#B794F4",
    400: "#9F7AEA",
    500: "#805AD5",
    600: "#6B46C1",
    700: "#553C9A",
    800: "#44337A",
    900: "#322659",
  },

  pink: {
    50: "#FFF5F7",
    100: "#FED7E2",
    200: "#FBB6CE",
    300: "#F687B3",
    400: "#ED64A6",
    500: "#D53F8C",
    600: "#B83280",
    700: "#97266D",
    800: "#702459",
    900: "#521B41",
  },
};

const namedColors = {
  variable: {
    main: baseColors.green[300],
    dark: baseColors.green[400],
    light: baseColors.green[200],
  },
  literals: {
    main: baseColors.teal[300],
    dark: baseColors.teal[400],
  },
};

module.exports = {
  ...baseColors,
  ...namedColors,
  current: "currentColor",
  alpha: (color, alpha) => `${color}${alpha}`,
};


scripts/components/colors.js
const palette = require("../palette");
module.exports = {
  "activityBar.background": palette.gray[1000],
  "activityBar.foreground": palette.gray[200],
  "activityBarBadge.background": palette.blue[600],
  "activityBarBadge.foreground": palette.blue[50],
  "badge.background": palette.gray[800],
  "button.background": palette.gray[600],
  "button.secondaryBackground": palette.gray[700],
  "button.secondaryForeground": palette.gray[100],
  "checkbox.border": palette.gray[600],
  "debugToolBar.background": palette.gray[1000],
  descriptionForeground: palette.gray[400],
  "diffEditor.insertedTextBackground": palette.alpha(palette.green[700], "30"),
  "diffEditor.diagonalFill": palette.gray[700],
  "dropdown.background": palette.gray[1000],
  "dropdown.border": palette.gray[1000],
  "editor.background": palette.gray[900],
  "editor.findMatchBackground": palette.alpha(palette.cyan[500], "50"),
  "editor.findMatchBorder": palette.cyan[500],
  "editor.findMatchHighlightBackground": palette.alpha(palette.cyan[500], "50"),
  "editor.foreground": palette.gray[400],
  "editorBracketHighlight.foreground1": palette.pink[500],
  "editorBracketHighlight.foreground2": palette.blue[500],
  "editorBracketHighlight.foreground3": palette.orange[500],
  "editorBracketHighlight.foreground4": palette.cyan[500],
  "editorBracketHighlight.foreground5": palette.yellow[500],
  "editorBracketHighlight.foreground6": palette.purple[500],
  "editorBracketHighlight.unexpectedBracket.foreground": palette.red[500],
  "editorHoverWidget.highlightForeground": palette.blue[300],
  "editorInlayHint.foreground": palette.gray[400],
  "editorInlayHint.background": palette.gray[700],
  "editor.lineHighlightBackground": palette.alpha(palette.purple[900], "80"),
  "editorLineNumber.activeForeground": palette.gray[400],
  "editorGutter.addedBackground": palette.green[500],
  "editorGutter.deletedBackground": palette.red[700],
  "editorGutter.modifiedBackground": palette.blue[400],
  "editorGutter.background": palette.gray[1000],
  "editorOverviewRuler.addedBackground": palette.green[500],
  "editorOverviewRuler.deletedBackground": palette.red[700],
  "editorOverviewRuler.modifiedBackground": palette.blue[400],
  "editorOverviewRuler.findMatchForeground": palette.cyan[500],
  "editor.selectionBackground": palette.alpha(palette.purple[500], "50"),
  "editor.selectionHighlightBackground": palette.transparent,
  "editor.selectionHighlightBorder": palette.gray[400],
  "editor.wordHighlightBackground": palette.transparent,
  "editor.wordHighlightBorder": palette.purple[300],
  "editor.wordHighlightStrongBackground": palette.transparent,
  "editor.wordHighlightStrongBorder": palette.purple[300],
  "editorActiveLineNumber.foreground": palette.gray[600],
  "editorBracketMatch.background": palette.alpha(palette.purple[500], "80"),
  "editorBracketMatch.border": palette.purple[500],
  "editorCursor.background": palette.alpha(palette.white[100], "10"),
  "editorCursor.foreground": palette.pink[400],
  "editorError.foreground": palette.red[500],
  "editorError.background": palette.alpha(palette.red[800], "80"),
  "editorGroup.background": palette.gray[1000],
  "editorGroup.border": palette.gray[1000],
  "breadcrumb.background": palette.gray[1000],
  "editorGroupHeader.tabsBackground": palette.gray[1000],
  "editorHoverWidget.background": palette.gray[900],
  "editorHoverWidget.border": palette.gray[900],
  "editorIndentGuide.activeBackground": palette.purple[600],
  "editorIndentGuide.background": palette.gray[600],
  "editorLineNumber.foreground": palette.gray[600],
  "editorMarkerNavigation.background": palette.gray[900],
  "editorRuler.foreground": palette.gray[700],
  "editorSuggestWidget.background": palette.gray[900],
  "editorSuggestWidget.border": palette.gray[900],
  "editorSuggestWidget.selectedBackground": palette.purple[600],
  "editorWarning.foreground": palette.yellow[600],
  "editorWhitespace.foreground": palette.gray[700],
  "editorWidget.background": palette.gray[900],
  focusBorder: palette.gray[600],
  "gitDecoration.ignoredResourceForeground": palette.gray[400],
  "input.background": palette.gray[800],
  "list.activeSelectionBackground": palette.gray[700],
  "list.activeSelectionForeground": palette.gray[200],
  "list.focusBackground": palette.gray[700],
  "list.focusForeground": palette.gray[200],
  "list.highlightForeground": palette.gray[200],
  "list.hoverBackground": palette.gray[700],
  "list.hoverForeground": palette.white,
  "list.inactiveSelectionBackground": palette.gray[900],
  "list.inactiveSelectionForeground": palette.gray[200],
  "list.warningForeground": palette.yellow[400],
  "menu.foreground": palette.gray[400],
  "menu.separatorBackground": palette.gray[700],
  "minimapGutter.addedBackground": palette.green[500],
  "minimapGutter.deletedBackground": palette.red[700],
  "minimapGutter.modifiedBackground": palette.blue[400],
  "minimap.findMatchHighlight": palette.cyan[700],
  "minimap.selectionHighlight": palette.alpha(palette.purple[500], "50"),
  "panel.border": palette.gray[600],
  "panelSectionHeader.background": palette.gray[900],
  "peekViewEditor.background": palette.gray[800],
  "peekViewEditor.matchHighlightBackground": palette.alpha(
    palette.cyan[900],
    "40"
  ),
  "peekViewResult.background": palette.gray[1000],
  "scrollbar.shadow": palette.gray[1000],
  "scrollbarSlider.activeBackground": palette.alpha(palette.gray[500], "80"),
  "scrollbarSlider.background": palette.alpha(palette.gray[500], "40"),
  "scrollbarSlider.hoverBackground": palette.alpha(palette.gray[500], "60"),
  "settings.focusedRowBackground": palette.gray[800],
  "settings.headerForeground": palette.white,
  "sideBar.background": palette.gray[1000],
  "sideBar.foreground": palette.gray[400],
  "sideBarSectionHeader.background": palette.gray[800],
  "sideBarSectionHeader.foreground": palette.gray[400],
  "statusBar.background": palette.gray[1000],
  "statusBar.debuggingBackground": palette.orange[500],
  "statusBar.debuggingBorder": palette.orange[500],
  "statusBar.debuggingForeground": palette.white,
  "statusBar.foreground": palette.gray[400],
  "statusBar.noFolderBackground": palette.gray[900],
  "statusBarItem.hoverBackground": palette.gray[700],
  "statusBarItem.remoteBackground": palette.blue[600],
  "statusBarItem.remoteForeground": palette.blue[50],
  "tab.activeBackground": palette.gray[900],
  "tab.activeForeground": palette.gray[200],
  "tab.border": palette.gray[1000],
  "tab.hoverBackground": palette.gray[700],
  "tab.inactiveBackground": palette.gray[1000],
  "tab.unfocusedHoverBackground": palette.gray[700],
  "terminal.ansiBlack": palette.black,
  "terminal.ansiBlue": palette.blue[400],
  "terminal.ansiBrightBlack": palette.gray[600],
  "terminal.ansiBrightBlue": palette.blue[300],
  "terminal.ansiBrightCyan": palette.cyan[300],
  "terminal.ansiBrightGreen": palette.green[300],
  "terminal.ansiBrightMagenta": palette.purple[300],
  "terminal.ansiBrightRed": palette.red[300],
  "terminal.ansiBrightWhite": palette.gray[200],
  "terminal.ansiBrightYellow": palette.yellow[300],
  "terminal.ansiCyan": palette.cyan[400],
  "terminal.ansiGreen": palette.green[400],
  "terminal.ansiMagenta": palette.purple[400],
  "terminal.ansiRed": palette.red[400],
  "terminal.ansiWhite": palette.gray[50],
  "terminal.ansiYellow": palette.yellow[400],
  "terminal.background": palette.gray[900],
  "terminal.border": palette.gray[600],
  "terminal.foreground": palette.gray[400],
  "terminal.selectionBackground": palette.alpha(palette.purple[500], "50"),
  "textLink.foreground": palette.blue[300],
  "titleBar.activeBackground": palette.gray[900],
  "titleBar.activeForeground": palette.gray[200],
  "titleBar.inactiveBackground": palette.gray[900],
  "titleBar.inactiveForeground": palette.gray[500],
  "welcomePage.buttonHoverBackground": palette.gray[600],
  "quickInput.background": palette.gray[1000],
};
